tests/activity-report.spec.ts

```diff
@@ -685,4 +685,66 @@ test.describe('Activity Report', () => {
     await expect(page.getByRole('link', { name: `R0${regionNumber}-AR-${arNumber}` })).toBeVisible();
   });
 
+  test('can remove objective', async ({ page }) => {
+    await getFullName(page);
+
+    await page.getByRole('link', { name: 'Activity Reports' }).click();
+    await page.getByRole('button', { name: '+ New Activity Report' }).click();
+
+    await activitySummary(page);
+
+    await page.getByRole('button', { name: 'Save and continue' }).click();
+
+    await page.getByRole('button', { name: 'Supporting attachments not started' }).click();
+    await page.getByRole('button', { name: 'Goals and objectives not started' }).click();
+
+    // create the first goal
+    await page.getByTestId('label').locator('div').filter({ hasText: '- Select -' }).nth(2)
+      .click();
+    await page.locator('#react-select-15-option-0').getByText('Create new goal').click();
+    await page.getByTestId('textarea').click();
+    await page.getByTestId('textarea').fill('g1');
+    await page.getByText('RTTAPA', { exact: true }).click();
+    await page.getByRole('button', { name: 'Save goal' }).click();
+
+    // create first objective
+    await page.locator('.css-125guah-control > .css-g1d714-ValueContainer').click();
+    await page.keyboard.press('Enter');
+    await page.getByLabel('TTA objective *').click();
+    await page.getByLabel('TTA objective *').fill('g1 o1 title');
+    await page.locator('.css-125guah-control > .css-g1d714-ValueContainer').click();
+    await page.locator('#react-select-21-option-0').click();
+    await blur(page);
+    await page.getByRole('textbox', { name: 'TTA provided for objective' }).locator('div').nth(2).click();
+    await page.keyboard.type('g1 o1 tta');
+    await blur(page);
+
+    // create second objective
+    await page.getByRole('button', { name: 'Add new objective' }).click();
+    await page.locator('[id="goalForEditing\\.objectives\\[1\\]\\.title"]').fill('g1 o2 title');
+    await blur(page);
+    await page.locator('[id="goalForEditing\\.objectives\\[1\\]\\.topics"]').focus();
+    await page.keyboard.press('ArrowDown');
+    await page.keyboard.press('Enter');
+    await blur(page);
+
```

Comment:
I think that for this test to be accurate to the issue you'd need to save, and **then** remove the objective, then refresh and verify objective count